### ISSUE TITLE
Defaults for bin counts

### ISSUE THREAD

**Issue by yurivish** [observablehq/plot#363]
Filing this issue at Mike's suggestion in the d3 slack. I ran into some surprising behavior using `Plot.binX` without specifying any arguments (https://observablehq.com/d/cf5ba27c74172e3e) and Mike suggested adding a default to handle this case.

Right now this produces the expected plot:
```js
Plot.rectY(data, Plot.binX({ y: 'count' })).plot()
```

While this produces an unexpected (not empty, but also not correct) plot:
```js
Plot.rectY(data, Plot.binX()).plot()
```

One question would be what channels to default to `count` – in this case I wanted the `y` channel, but I'm not sure if that's always the right default.

**Comment by Fil** [comment]
What happens here is that we have the rule [in Plot.rectY](https://github.com/observablehq/plot#plotrectydata-options): *if the y option is not specified, it defaults to the identity function.*

this turns Plot.rectY(data, Plot.binX()).plot() into the equivalent of:
```js
Plot.rectY(
  data,
  Plot.binX({ y: (d) => d })
).plot()
```

now the identity function is taken as the y *reducer*, and so returns,  for each bin, the array of values contained in that bin: [d0, d1, d2…].

in the stack function, we have:
```js
Y = valueof(data = [bins], y = (lazytransform…), Float64Array)
```

This in turn calls something like Float64Array.from([[0, 1], [2]]), which results in [NaN, 2].
In other words when a bin contains just 1 value [d0], the value is taken as d0. If it contains more values [d0, d1…], it is NaN. So the vertical bars you see are the bins that contain exactly one data point.

**Comment by Fil** [comment]
Going through this again; the bug is provoked by the following sequence:

1) The grouping operation has (correctly) transformed the data into 5 buckets, say [[0.01, 0.12, 0.19…], [0.21, 0.22, 0.39…], [0.41, 0.52, 0.59…], [0.61, 0.72, 0.79…], [0.84]]. Note how only the last of these buckets has 1 element.

2) The next transform is the implicit Plot.stackY transform called by Plot.rectY. It receives our list of buckets as *data* (an array of arrays), and applies:
```
Y = valueof(data, y, Float64Array)
```
with `y` being identity (the default for Plot.stackY).

This by the magic virtue of JavaScript's typecasting returns a Float64Array(NaN, NaN, NaN, NaN, NaN, 0.84). Et voilà, we get 1 bar of height 0.84.


The fix is simply (as you suggested) to add correct defaults in binX: `binX(outputs = {y: "count"}, …`

I'll add a test for binX, binY, and probably bin also needs a default output ({r: "count"} ?)
